src/repart/repart.c

```diff
@@ -6017,7 +6017,8 @@ static int do_copy_files(Context *context, Partition *p, const char *root) {
                 if (r < 0)
                         return r;
 
-                sfd = chase_and_open(line->source, arg_copy_source, CHASE_PREFIX_ROOT, O_CLOEXEC|O_NOCTTY, NULL);
+                int source_is_symlink = is_symlink(line->source);
```

Comment:
so you are not considering arg_copy_source here. moreover, the whole excercise of chase_and_open() is to pin the inodes once (i.e. derive inode from path) and then operate exclusively with that. Hence, maybe use chase() here first, to obtain, an O_PATH fd to the inode, then make your decisions, and eventually fd_reopen() it to get a real fd off the same fd.